src/bitcoin-cli.cpp

```diff
@@ -474,6 +511,46 @@ static int CommandLineRPC(int argc, char *argv[])
         if (gArgs.GetBoolArg("-getinfo", false)) {
             rh.reset(new GetinfoRequestHandler());
             method = "";
+        } else if (gArgs.GetBoolArg("-generate", false)) {
+            //-generate is special as it executes two RPC calls:
+            // 1) "getnewaddress" without any parameters, which returns a bech32 address
+            // 2) "generatetoaddress" with three params: numblocks, address and maxtries
+            // If no args are given the args-vector will be extended with DEFAULT_GENERATED_NBLOCKS
+            // and DEFAULT_GENERATED_MAXTRIES.
+            // In any case, the address will be generated internally and put into args vector as well.
+            rh.reset(new GetNewAddressRequestHandler());
+            method = "";
+            const UniValue reply = CallRPC(rh.get(), method, args);
+            const UniValue& result = find_value(reply, "result");
+            const UniValue& error  = find_value(reply, "error");
+            if (error.empty()) {
+                const std::string address = result.get_obj()["result"].getValStr();
+                // Take care of different situations:
+                // a) no arguments given
+                // b) only "nblocks"
+                // c) "nblocks" & "maxtries"
+                // --------------------------
+                // "address" must be the 2nd arg in the vector before calling generatetoaddress.
+                if (args.size() == 0) {
+                    args.push_back(std::to_string(DEFAULT_GENERATED_NBLOCKS));
+                    args.push_back(std::move(address));
+                }
+                else if (args.size() < 2) {
+                    args.push_back(std::move(address));
+                }
+                else if (args.size() == 2) {
+                    const std::string nblocks = args[0];
+                    const std::string maxtries = args[1];
```

Comment:
style-nit: I prefer `.at(0)` and `.at(1)`. Feel free to ignore.

Reply:
**at** makes more sense, in every way. Will adapt.